File: VS2005/CS/BindingFiltering/BindingFiltering/Employee.cs
//---------------------------------------------------------------------
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//---------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Text;

namespace BindingFiltering
{
    public class Employee
    {
        private string lastNameValue;
        private string firstNameValue;
        private int salaryValue;
        private DateTime startDateValue;
        public Employee()
        {
            LastName = "Last Name";
            FirstName = "First Name";
            Salary = 40000;
            StartDate = DateTime.Today;
        }
        public Employee(string lastName, string firstName,
            int salary, DateTime startDate)
        {
            LastName = lastName;
            FirstName = firstName;
            Salary = salary;
            StartDate = startDate;
        }

        public string LastName
        {
            get { return lastNameValue; }
            set { lastNameValue = value; }
        }
        public string FirstName
        {
            get { return firstNameValue; }
            set { firstNameValue = value; }
        }

        public int Salary
        {
            get { return salaryValue; }
            set { salaryValue = value; }
        }

        public DateTime StartDate
        {
            get { return startDateValue; }
            set { startDateValue = value; }
        }

        public override string ToString()
        {
            return LastName + ", " + FirstName + "\n" +
                "Salary:" + salaryValue + "\n" +
                "Start date:" + startDateValue;
        }

    }
}

File: VS2005/VB/BindingFiltering/BindingFiltering/Employee.vb
'---------------------------------------------------------------------
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'---------------------------------------------------------------------

Option Explicit On
Option Strict On

Public Class Employee

    Private lastNameValue As String
    Private firstNameValue As String
    Private salaryValue As Integer
    Private startDateValue As DateTime

    Public Sub New()
        Me.LastName = "Last Name"
        Me.FirstName = "First Name"
        Me.Salary = 40000
        Me.StartDate = DateTime.Today
    End Sub

    Public Sub New(ByVal lastName As String, ByVal firstName As String, _
    ByVal salary As Integer, ByVal startDate As DateTime)
        Me.LastName = lastName
        Me.FirstName = firstName
        Me.Salary = salary
        Me.StartDate = startDate
    End Sub

    Public Property LastName() As String
        Get
            Return lastNameValue
        End Get
        Set(ByVal value As String)
            lastNameValue = value
        End Set
    End Property

    Public Property FirstName() As String
        Get
            Return firstNameValue
        End Get
        Set(ByVal value As String)
            firstNameValue = value
        End Set
    End Property

    Public Property Salary() As Integer
        Get
            Return salaryValue
        End Get
        Set(ByVal value As Integer)
            salaryValue = value
        End Set
    End Property

    Public Property StartDate() As DateTime
        Get
            Return startDateValue
        End Get
        Set(ByVal value As DateTime)
            startDateValue = value
        End Set
    End Property

    Public Overrides Function ToString() As String
        Return LastName + ", " + FirstName + vbCrLf + _
            "Salary:" + salaryValue.ToString() + vbCrLf + _
            "Start date:" + startDateValue.ToString()
    End Function

End Class
